File: PHP/lambda/1/main.php
<?php
/**
 * Created by IntelliJ IDEA.
 * User: fay
 * Date: 2017/5/31
 * Time: 下午3:47
 */

    function main() {
        say(function ($v) {
            echo $v;
            return "Hello PHP!\n";
        });
    }

    function say($lambda) {
        echo $lambda("Hello World!\n");
    }

    main();

File: Scala/lambda/1/main.scala
/**
  * Created by fay on 2017/6/1.
  */

object main {
  def main(args: Array[String]): Unit = {
    say(string => {
      println(string)
      "Hello Scala!"
    })
  }

  def say(lambda: (String) => String) = {
    println(lambda("Hello World!"))
  }
}
